import/popups/PopupOpener.qml

```diff
@@ -184,6 +185,26 @@ Timer {
             root._openContextMenu(data)
             break
         }
+        case "embed:webrtcrequest": {
+            if (data.title === "microphone") {
+                var obj = pageStack.animatorPush(Qt.resolvedUrl("MicrophoneDialog.qml"), {"host": data.host })
```

Comment:
Using the test page, if I click the "Microphone" button in quick succession it can trigger two `embed:webrtcrequest` requests before the page has had time to push. It results in the following error:
```
JSComp: EmbedLiteWebrtcUI.js: Created request camera:{02190832-072d-4d90-b994-8bea23633cf1}
[D] doPush:134 - Warning: cannot push while transition is in progress
[W] unknown:191 - file:///usr/lib/qt5/qml/Sailfish/WebView/Popups/PopupOpener.qml:191: TypeError: Cannot read property 'pageCompleted' of null
JSComp: EmbedLiteWebrtcUI.js: Allow video device -1
```
This causes a bit of a mess because it means `embedui:webrtcresponse` is never sent, so `WebrtcPermissionRequest` is never resolved, and so most importantly `Services.embedlite.removeMessageListener("embedui:webrtcresponse", this)` is never called in `EmbedLiteWebrtcUI._prompt()`. Future calls therefore result in multiple calls to `EmbedLiteWebrtcUI.onMessageReceived()` causing an error. See the following output:

```
JSComp: EmbedLiteWebrtcUI.js: Audio devies 6 video devices 0
JSComp: EmbedLiteWebrtcUI.js: Created request microphone:{481c2e96-ea9c-4bd2-9809-2d5985fe9df5}
JSComp: EmbedLiteWebrtcUI.js: Submitting request microphone id microphone:{481c2e96-ea9c-4bd2-9809-2d5985fe9df5}
JSComp: EmbedLiteWebrtcUI.js: Created request camera:{481c2e96-ea9c-4bd2-9809-2d5985fe9df5}
JSComp: EmbedLiteWebrtcUI.js: Allow video device -1
JSComp: EmbedLiteWebrtcUI.js: onMessageReceived: top:embedui:webrtcresponse, msg:{"allow":true,"checkedDontAsk":false,"id":"microphone:{481c2e96-ea9c-4bd2-9809-2d5985fe9df5}"}
JSComp: EmbedLiteWebrtcUI.js: onMessageReceived: top:embedui:webrtcresponse, msg:{"allow":true,"checkedDontAsk":false,"id":"microphone:{481c2e96-ea9c-4bd2-9809-2d5985fe9df5}"}
JSComp: EmbedLiteWebrtcUI.js: Allow audio device 0
JSComp: EmbedLiteWebrtcUI.js: Selected audio device: default: Null Input
JSComp: EmbedLiteWebrtcUI.js: getUserMedia:response:allow for callID {481c2e96-ea9c-4bd2-9809-2d5985fe9df5}
JSComp: EmbedLiteWebrtcUI.js: got getUserMedia:request
JavaScript error: file:///usr/lib/mozembedlite/components/EmbedLiteWebrtcUI.js, line 225: NS_ERROR_UNEXPECTED: 
CONSOLE message:
[JavaScript Error: "NS_ERROR_UNEXPECTED: " {file: "file:///usr/lib/mozembedlite/components/EmbedLiteWebrtcUI.js" line: 225}]
```

Reply:
Just an important addendum: I added some extra lines to the code for debug output, so the line number for that javascript is 219 in [this version](https://git.sailfishos.org/mer-core/embedlite-components/merge_requests/120/diffs) of the code.

Reply:
Thank you for review. The problem with pagestack transition is fixed.  I rewrote the dialog to support cameras as well. This depends on the new version of https://git.sailfishos.org/mer-core/embedlite-components/merge_requests/120.